### ISSUE TITLE
Query.node defined in resolvers, but not in schema

### ISSUE THREAD

**Issue by GreenOn** [gatsbyjs/gatsby#25405]
## Description

We are trying to make use of an existing Apollo Graphql Server for a new Gatsby site. I am not sure why but we are getting this error on `gatsby develop`.

Error: `Query.node defined in resolvers, but not in schema`. 

Our Apollo server has been working with no issues and the above error we are getting from Gatsby doesn't seem to be legitimate.

Our schema does have the `node` definition as shown below:

```
interface Node {
  id: ID!
}
type Query{
  .....
  ...
  node(id: ID!): Node
}
```
I enabled the playground for the graphql server also if you need any additional info.

### Steps to reproduce

Here is a simple repo you can use to reproduce the problem: [https://github.com/GreenOn/gatsby-apollo-dynos](https://github.com/GreenOn/gatsby-apollo-dynos). 

I followed the example posted here and updated the config file with our graphql server name: [https://github.com/jlengstorf/gatsby-with-apollo](https://github.com/jlengstorf/gatsby-with-apollo)

### Expected result

The expected result is to run the project successfully getting all the queries working at the build time.

### Actual result
On executing `gatsby develop`, it fails immediately after loading `gatsby-config.js`.

### Environment
  System:
    OS: Windows 10 10.0.18362
    CPU: (4) x64 Intel(R) Core(TM) i7-6500U CPU @ 2.50GHz
  Binaries:
    Node: 12.16.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.4 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.14.0 - C:\Program Files\nodejs\npm.CMD
  Languages:
    Python: 2.7.16 - /c/Users/pramo/.windows-build-tools/python27/python
  Browsers:
    Edge: 44.18362.449.0
  npmPackages:
    gatsby: ^2.0.19 => 2.23.11 
    gatsby-source-graphql: ^2.0.6 => 2.5.7 
  npmGlobalPackages:
    gatsby-cli: 2.12.50

**Comment by vladar** [comment]
Quick googling shows that it is probably an apollo-server issue (at least there are similar reports in their repo a couple of times):

https://github.com/apollographql/apollo-server/issues/2656
https://github.com/apollographql/apollo-server/issues/1534

Have you checked those?

**Comment by GreenOn** [comment]
Yes, but why only Gatsby graphql plugin complains about this? Our graphql server has been working fine for some time. If we have to make changes to get this running, we can, but I find it odd that only Gatsby has issues with this?

**Comment by vladar** [comment]
You are right, it happens when we try to transform the schema using `graphql-tools`. Somewhere in this call:

https://github.com/gatsbyjs/gatsby/blob/c2f42cbaf30ae1cdc69ab800c59ba91d5e092e29/packages/gatsby-source-graphql/src/gatsby-node.js#L97-L101

It throws even without custom Gatsby transforms. So probably a bug somewhere in `graphql-tools`.

One interesting thing though is that you have this `Node` interface in your schema but there is no single type implementing this interface. My best bet is that it confuses `graphql-tools` somehow.

Do you even use `Query.node`? Looks like it can't return anything, can it?

**Comment by GreenOn** [comment]
I actually had to convert the `interface` to `type` to get this working. Actually, this whole schema was auto-generated by Prisma (v1).
It has a whole bunch of things like `interface`, `input`, `DateTime` etc. I tried to upgrade `graphql-tools` to the [latest v6](https://www.graphql-tools.com/docs/migration-from-tools-v5), but it is throwing more errors now on the schema generated by Prisma. So for now, I took this route and got this working.  Yes, there are so many moving parts here and everything needs to be in sync to get it working.
